realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -85,45 +91,66 @@ static RealmCacheType valueOf(Class<? extends BaseRealm> clazz) {
 
     // Realm path will be used as the key to store different RealmCaches. Different Realm configurations with same path
     // are not allowed and an exception will be thrown when trying to add it to the cache map.
-    private static final Map<String, RealmCache> cachesMap = new HashMap<>();
+    // A weak ref is used to hold the RealmCache instance. The weak ref entry will be cleared if and only if there
+    // is no Realm instance holding a strong ref to it and there is no Realm instance associated it is BEING created.
+    private static final Map<String, WeakReference<RealmCache>> cachesMap =
```

Comment:
I don't think we can use weak reference here.
If a user lost all reference to a Realm instance without calling `close()`,  corresponding `RealmCache` instance seems to be GCed since the strong reference to the `RealmCache`is only exists in the Realm instance.